src/test/libslic3r/test_thin.cpp

```diff
@@ -0,0 +1,470 @@
+    //    } };
+    //    expolygon.contour.make_counter_clockwise();
+    //    WHEN("creating the medial axis"){
+    //        Polylines res;
+    //        expolygon.medial_axis(scale_(1), scale_(0.25), &res);
+    //        THEN("medial axis of a tooth is two lines"){
+    //            REQUIRE(res.size() == 2);
+    //            THEN("medial axis has reasonable length") {
+    //                REQUIRE(res[0].length() >= scale_(1.4) - SCALED_EPSILON);
+    //                REQUIRE(res[1].length() >= scale_(1.4) - SCALED_EPSILON);
+    //                // TODO: check if min width is < 0.3 and max width is > 0.6 (min($res->[0]->width.front, $res->[0]->width.back) # problem: can't have access to width
+    //                //TODO: now i have access! correct it!
+    //            }
+    //        }
+    //    }
+    //} 
+
+    GIVEN("Anchor & Tapers")
+    {
+        ExPolygon tooth;
+        tooth.contour = Slic3r::Polygon{ Points{
+            Point::new_scale(0,0), Point::new_scale(10,0), Point::new_scale(10,1.2), Point::new_scale(0,1.2)
+        } };
+        tooth.contour.make_counter_clockwise();
+        ExPolygon base_part;
+        base_part.contour = Slic3r::Polygon{ Points{
+            Point::new_scale(0,-3), Point::new_scale(0,3), Point::new_scale(-2,3), Point::new_scale(-2,-3)
+        } };
+        base_part.contour.make_counter_clockwise();
+        //expolygon.contour = Slic3r::Polygon{ Points{
+        //    //Point::new_scale(0, 13), Point::new_scale(-1, 13), Point::new_scale(-1, 0), Point::new_scale(0.0,0.0),
+        //    Point::new_scale(0,0.2), Point::new_scale(3,0.2), Point::new_scale(3,0.4), Point::new_scale(0,0.4),
+        //    Point::new_scale(0,1), Point::new_scale(3,1), Point::new_scale(3,1.3), Point::new_scale(0,1.3),
+        //    Point::new_scale(0,2), Point::new_scale(3,2), Point::new_scale(3,2.4), Point::new_scale(0,2.4),
+        //    Point::new_scale(0,3), Point::new_scale(3,3), Point::new_scale(3,3.5), Point::new_scale(0,3.5),
+        //    Point::new_scale(0,4), Point::new_scale(3,4), Point::new_scale(3,4.6), Point::new_scale(0,4.6),
+        //    Point::new_scale(0,5), Point::new_scale(3,5), Point::new_scale(3,5.7), Point::new_scale(0,5.7),
+        //    Point::new_scale(0,6), Point::new_scale(3,6), Point::new_scale(3,6.8), Point::new_scale(0,6.8),
+        //    Point::new_scale(0,7.5), Point::new_scale(3,7.5), Point::new_scale(3,8.4), Point::new_scale(0,8.4),
+        //    Point::new_scale(0,9), Point::new_scale(3,9), Point::new_scale(3,10), Point::new_scale(0,10),
+        //    Point::new_scale(0,11), Point::new_scale(3,11), Point::new_scale(3,12.2), Point::new_scale(0,12.2),
+        //} };
+        WHEN("1 nozzle, 0.2 layer height") {
+            const coord_t nozzle_diam = scale_(1);
+            ExPolygon anchor = union_ex(ExPolygons{ tooth }, intersection_ex(ExPolygons{ base_part }, offset_ex(tooth, nozzle_diam / 2)), true)[0];
```

Comment:
This version of `union_ex` doesn't seem to exist?

Reply:
For the x() it's something from prusa, i have to remove the () when converting for here.

I added this union. Basically, adding all expoly into a vector and returning the unionex of it.

Reply:
I guess we should check to see if wrapping it in an accessor is worth converting code too or if it was just something their fork did because reasons